### PR TITLE
A cleanup PR

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
This PR is about : 
- making the code easier to read
- simplified to over-complefixed logic
- fixing typos
- fixing some stylish issues
- removing useless files
- protecting the code against wrong behaviors
- refactoring shared code across files

### CODE REVIEW

**Comment by ktdreyer** [comment]
Thanks a lot for this PR

> 156 lines added, 892 lines removed

In the future it would be great to split unrelated changes up so they can be reviewed more safely / quickly.

**Comment by ErwanAliasr1** [comment]
Splitting commit or PR ?

File: ceph-releases/kraken/ubuntu/16.04/daemon/Dockerfile [hunk 1]
```diff
@@ -29,7 +29,7 @@ ADD https://bin.equinox.io/c/ekMN3bCZFUn/forego-stable-linux-amd64.tgz /forego.t
 
 
 # install prerequisites
-RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y wget unzip uuid-runtime python-setuptools udev runit sharutils python3 dmsetup && \
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -y wget unzip uuid-runtime python-setuptools udev sharutils python3 dmsetup && \
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/Dockerfile]: I think we might be able to remove sharutils and python3, let's test and see how the CI reacts.

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh [hunk 2]
```diff
@@ -45,22 +42,11 @@ function get_mon_config {
...
-    until confd -onetime -backend ${KV_TYPE} -node ${CONFD_NODE_SCHEMA}${KV_IP}:${KV_PORT} ${CONFD_KV_TLS} -prefix="/${CLUSTER_PATH}/" ; do
-      log "Waiting for confd to update templates..."
-      sleep 1
-    done
-
-    # Check/Create bootstrap key directories
-    mkdir -p /var/lib/ceph/bootstrap-{osd,mds,rgw}
-
-    log "Adding Keyrings"
+    log "Adding mon/admin Keyrings"
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: why did you remove the until loop?

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh [hunk 3]
```diff
@@ -45,22 +42,11 @@ function get_mon_config {
...
     kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/monKeyring > /etc/ceph/${CLUSTER}.mon.keyring
-    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/adminKeyring > /etc/ceph/${CLUSTER}.client.admin.keyring
-    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/bootstrapOsdKeyring > /var/lib/ceph/bootstrap-osd/${CLUSTER}.keyring
-    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/bootstrapMdsKeyring > /var/lib/ceph/bootstrap-mds/${CLUSTER}.keyring
-    kviator --kvstore=${KV_TYPE} --client=${KV_IP}:${KV_PORT} ${KV_TLS} get ${CLUSTER_PATH}/bootstrapRgwKeyring > /var/lib/ceph/bootstrap-rgw/${CLUSTER}.keyring
-
+    get_admin_key
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: This is not enough, how do we get the other keys?

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-xenial_cluster

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh [hunk 4]
```diff
@@ -45,21 +45,33 @@ function create_socket_dir {
   chown ceph. /var/run/ceph
 }
 
+# create the bootstrap directories
+function create_bootstrap_directories{
+  for directory in osd mds rgw; do
+    mkdir -p /var/lib/ceph/bootstrap-$directory
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: Can we enhance this one with `/var/lib/ceph/{osd,mon,mds,rgw,tmp}` as well?
So we can use it more easily and elsewhere in the code.

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 2 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: because get_config does it, that was duplicated code from there

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 3 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: already done by get_config, this part is just removing the actual code by what is done by get_config.

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh (hunk 4 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: sure will do it

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: Thanks!

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 3 above)
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: ack =)

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 2 above)
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: ack =)

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh (hunk 4 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: I only find a new occurences of mkdir -p /var/lib/ceph/osd, not any other.
Does it worth moving all of them into that function ?

File: ceph-releases/kraken/ubuntu/16.04/daemon/Dockerfile (hunk 1 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/Dockerfile]: ok done.

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh (hunk 4 above)
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: Yes please, this won't hurt :)

Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: Why not /var/run/ceph by the way ?

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: Good point, we create this one as well. Please add it :)

**Comment by ktdreyer** [comment]
> Splitting commit or PR ?
Splitting the PR

**Comment by leseb** [comment]
jenkins test this please

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh [hunk 5]
```diff
@@ -39,27 +39,45 @@ function prefix_length {
   done
 }
 
-# create socket directory
-function create_socket_dir {
+
+# create the mandatory directories
+function create_mandatory_directories{
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: you need a space after `create_mandatory_directories`. Otherwise the function is not happy.

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh [hunk 6]
```diff
@@ -6,62 +6,63 @@ function osd_disk_prepare {
...
+    DATA_UUID=$(blkid -o value -s PARTUUID ${OSD_DEVICE}1)
+    umount /var/lib/ceph/osd-lockbox/${DATA_UUID} || true
+  else
+    ceph-disk -v prepare ${CEPH_OPTS} --journal-uuid ${OSD_JOURNAL_UUID} ${OSD_DEVICE} ${OSD_JOURNAL}
+  fi
+
+  if [[ -n "${OSD_JOURNAL}" ]]; then
     chown ceph. $(dev_part ${OSD_DEVICE} 2)
+  else
+    chown ceph. ${OSD_JOURNAL}
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: This `else` can not work.
`-n` is true if the length of string is nonzero.

In the case it's False, we enter the `else` but `OSD_JOURNAL` is empty so this fails.

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: We should use `-z` instead.

File: ceph-releases/kraken/ubuntu/16.04/daemon/entrypoint.sh [hunk 7]
```diff
@@ -31,6 +30,8 @@ esac
 # Normalize DAEMON to lowercase
 CEPH_DAEMON=$(echo ${CEPH_DAEMON} |tr '[:upper:]' '[:lower:]')
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/entrypoint.sh]: Looks like there is an unnecessary space before that. Can you remove it?

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh [hunk 8]
```diff
@@ -141,11 +124,10 @@ function get_config {
   done
 
   # Check/Create bootstrap key directories
-  mkdir -p /var/lib/ceph/bootstrap-{osd,mds,rgw}
+  create_bootstrap_directories
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: Do we still need this one since we have `create_mandatory_directories`?

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: We should remove it.

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh [hunk 9]
```diff
@@ -6,62 +6,63 @@ function osd_disk_prepare {
...
     else
-      ceph-disk -v prepare ${CEPH_OPTS} --journal-uuid ${OSD_JOURNAL_UUID} ${OSD_DEVICE}
+      log "INFO- It looks like ${OSD_DEVICE} is an OSD, set OSD_FORCE_ZAP=1 to use this device anyway and zap its content"
+      log "You can also use the zap_device scenario on the appropriate device to zap it"
+      log "Moving on, trying to activate the OSD now."
     fi
+  fi
+
+  if [[ ${OSD_BLUESTORE} -eq 1 ]]; then
+    ceph-disk -v prepare ${CEPH_OPTS} --bluestore ${OSD_DEVICE} ${OSD_JOURNAL}
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: Please remove `${OSD_JOURNAL}` from Bluestore.

File: ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh (hunk 5 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/common_functions.sh]: fixed.

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh (hunk 6 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: my bad

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 8 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: true.

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh (hunk 9 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: done

File: ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh (hunk 8 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/config.kv.sh]: done

File: ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh (hunk 6 above)
Comment (ErwanAliasr1) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: fixed

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_disk_prepare.sh]: Not fixed, you need `-z`.

**Comment by leseb** [comment]
jenkins test ceph_ansible-jewel-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-jewel-xenial_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-jewel-centos7_cluster

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-centos7_cluster

**Comment by leseb** [comment]
Merging this even if Jewel jobs are failing since this PR only affects Kraken.
